source/dbo/Stored Procedures/Unmatched Specimens/uspGenerateUnmatchedSpecimensLinkedNotifications.sql

```diff
@@ -0,0 +1,66 @@
+﻿CREATE PROCEDURE [dbo].[uspGenerateUnmatchedSpecimensLinkedNotifications]
+AS
+BEGIN TRY
+
+	-- First populate RejectSpecimenUsers table
+	EXEC dbo.uspGenerateRejectSpecimenUsers
+
+	TRUNCATE TABLE [dbo].[UnmatchedSpecimensLinkedNotifications]
+	
+	SELECT us.ReferenceLaboratoryNumber, nsm.NotificationID, 'Unmatch' AS EventType INTO #PreviousMatch
+	FROM UnmatchedSpecimens us
+	JOIN [$(NTBS_Specimen_Matching)].dbo.NotificationSpecimenMatch nsm ON nsm.ReferenceLaboratoryNumber = us.ReferenceLaboratoryNumber
+	WHERE nsm.MatchType = 'Rejected'
+
+	SELECT us.ReferenceLaboratoryNumber, nsm.NotificationID, 'RejectPotentialMatch' AS EventType INTO #PreviousPossibleMatch
+	FROM UnmatchedSpecimens us
+	JOIN [$(NTBS_Specimen_Matching)].dbo.NotificationSpecimenMatch nsm ON nsm.ReferenceLaboratoryNumber = us.ReferenceLaboratoryNumber
+	WHERE nsm.MatchType = 'Rejected-Possible'
+
+	SELECT mr.clientsourceID AS ReferenceLaboratoryNumber, mr.rclientsourceID AS NotificationID, 'PoorQualityMatch' AS EventType INTO #PoorQualityMatch
+	FROM UnmatchedSpecimens us
+	JOIN [$(NTBS_Specimen_Matching)].dbo.MatchResults mr ON mr.clientsourceID = us.ReferenceLaboratoryNumber
+	--Exclude any matches that have previously been rejected (otherwise every rejection has an accompanying match)
+	LEFT JOIN #PreviousMatch pm on pm.NotificationID = mr.rclientsourceID AND pm.ReferenceLaboratoryNumber = mr.clientsourceID
```

Comment:
This isn't strictly necessary, but it prevents any rejected match from showing as both a rejection and a poor quality match in the report